src/OrphanFileHelper.js

```diff
@@ -0,0 +1,82 @@
+const fs = require('fs');
+
+const logger = require('./logging');
+const fileStore = require('./store/file');
+
+/**
+ *
+ */
+class OrphanFileHelper {
+    /**
+     *
+     */
+    constructor({ tempPath, reconnectTimeout, dumpPersister }) {
+        this.tempPath = tempPath;
+        this.reconnectTimeout = reconnectTimeout;
+        this.dumpPersister = dumpPersister;
+    }
+
+    /**
+     * Remove old files from the temp folder.
+     */
+    processOldFiles() {
+        logger.info('[OrphanFileHelper] Waiting for connections to reconnect.');
+        try {
+            if (fs.existsSync(this.tempPath)) {
+                fs.readdirSync(this.tempPath).forEach(fname => {
+                    try {
+                        const filePath = `${this.tempPath}/${fname}`;
+
+                        logger.debug(`[OrphanFileHelper] Trying to process file ${filePath}`);
+                        fs.stat(filePath, (err, stats) => {
+                            if (err) {
+                                throw err;
```

Comment:
I think fs.stat is async which means that the exception will get send to the global unhandled exception handler, from what I can tell you want exceptions here to be contained, you could just log it instead of re-throwing.

Reply:
Like this?